### ISSUE TITLE
Some blocks not exporting

### ISSUE THREAD

**Issue by TheLimeHit** [jmc2obj/j-mc-2-obj#175]
Hello!
I recently noticed when I exported my build, that gates and trapdoors did not export with it. I first thought that it was an individual case, exported it again and got the same result. I tried changing different settings, using the default texture pack instead of the custom one, but nothing helped.
Another thing I noticed is that some trapdoors did export, but most still did not (very clearly seen on the windows of the building), and I don't know what makes the program remove the missing blocks.
The console does not mention any related blocks not found in the texture pack (just the orange glazed terracotta3 that wasn't even used in the building).

Using the v112 version helped, however many 1.17 blocks were used on the building, as well as player heads, so using v112 for me is not a good option, and overall the origin of this issue is unknown at least to me.
Thank you.

![2021-11-09 033232](https://user-images.githubusercontent.com/93954796/140841866-7c57386c-1201-438b-afcf-9815a5043694.jpg)
![2021-11-09 033259](https://user-images.githubusercontent.com/93954796/140841869-0a939448-5b46-4bd9-a7cc-72797ea33e35.jpg)
![2021-11-09 035329](https://user-images.githubusercontent.com/93954796/140841875-60ef1d5b-ad26-41a6-9345-6d60a4738712.jpg)
![2021-11-09 034753](https://user-images.githubusercontent.com/93954796/140841882-c8d8cf6f-c27c-4d21-904c-06ee3ad916c0.jpg)
![2021-11-09 035306](https://user-images.githubusercontent.com/93954796/140841888-00292ba9-9779-4ce3-9378-9946ace2b68b.jpg)
![2021-11-09_00 22 06](https://user-images.githubusercontent.com/93954796/140841897-55e6bc79-a2f2-4c72-8310-b88817cba518.png)

**Comment by mmdanggg2** [comment]
Trapdoors seem to be working fine for me, are they a specific type of trapdoor? Also make sure the blocks are enabled in the list: 
![image](https://user-images.githubusercontent.com/3161587/141030069-b60cc2bd-a88e-4375-8e9a-2d32723fa599.png)

**Comment by TheLimeHit** [comment]
Yes, I checked the list every time and did "Select All", and also made sure I did the same this time.

Those are birch trapdoors and jungle gates disappearing on that particular build.

I also noticed that it's only the old wooden (oak, birch, dark oak, jungle etc), trapdoors disappearing, as warped trapdoors (the grey ones on the building I originally shared above) and crimson trapdoors (a small grey line on the roof seen in the 5th picture of the above) exported without any problems. And the gates are all just disappearing, even the crimson and warped ones, for example here are birch ones missing, along with the birch trapdoors.

In-game screenshot (only added 3 blocks in height after that was taken, did zero changes in the details):
![2021-03-14_17 17 02](https://user-images.githubusercontent.com/93954796/141215960-c466ff0f-2175-4f43-9039-23b67127579f.png)

Side-by-side placed towers, v114 on the left and v112 on the right:
![2021-11-11 034615](https://user-images.githubusercontent.com/93954796/141216754-553f3684-2f94-4155-aa00-0dda6cc50e31.jpg)

A close-up of the v114-exported clock of the tower:
![2021-11-11 034029](https://user-images.githubusercontent.com/93954796/141216803-f84fe367-2414-4014-95ad-2ba9988b5837.jpg)

**Comment by mmdanggg2** [comment]
Still not seeing this issue with any of those blocks. Have you tried resetting the settings? delete the conf folder and reset to factory:
![image](https://user-images.githubusercontent.com/3161587/141225558-bd210891-14e0-49c5-a379-9b6e1070ac3d.png)

Also enable debug in the log, might show something else. You could also try and add 1.17.jar manually as a resource pack.

**Comment by TheLimeHit** [comment]
I deleted the conf folder, reset the settings, and it did not help with. The debug console showed this:
![2021-11-11 085510](https://user-images.githubusercontent.com/93954796/141290388-1a38aaa6-b76e-4ab2-9925-52ee85c481ce.jpg)

After the factory reset I decided to try the default pack first, and it exported without a single issue.
Next I decided to try the resource pack, but a different one, not mine. And it still worked.

I'm terribly sorry for raising a false alarm, it must've been that I messed something up when I was working on the new version of the resource pack. I'm assuming why the default pack didn't work the first try was because my pack could change up the config or something. But anyway, now I know that the resource pack is what's causing it.
 Thank you!

**Comment by mmdanggg2** [comment]
Those couldn't find errors are probably because of the missing conf folder, you just needed to restart jmc after deleting to recreate the defualt conf folder. If you have a resource pack that works in minecraft but not in jmc could you send that so I can see if it's fixable?

**Comment by TheLimeHit** [comment]
I can send it, yes. I'm a really beginner texture pack creator though, so I would really appreciate that :D. And yeah, it works perfectly in-game and it was fine to me, only noticed the problem when exported the build with it.
https://drive.google.com/file/d/1sC6-vav975S4l_oJsCoTgCHwPsUyLLKb/view?usp=sharing

**Comment by mmdanggg2** [comment]
Managed to figure this one out. I had made some incorrect assumptions about how the resource pack blockstates work in the game.

Jmc only used the first blockstate json file it found from the top most resource pack. So in your pack, the `birch_fence_gate` for example, only defined models for the `powered=true` state so none of them matched with the `powered=false` state of the blocks and jmc just didnt find a model to use. However, in minecraft, if there are no matching blockstate definitions it will then look for one in the next resource pack down, jmc will now do this too.

Your pack also had missing banner patterns so not all of them were high resolution which caused some banners to fail exporting, I've improved the banner code to handle banner patterns with differing resoultions.

Just one last thing I noteced with your pack specifically, the oak and acacia trapdoors are using the `template_plank_trapdoor_open` with invisible textures, which might cause issues, just FYI: 
![image](https://user-images.githubusercontent.com/3161587/141537335-190d9f47-d31a-4450-957c-285856842b1d.png)


Thanks for reporting, I got some good information from that!
